api/src/main/scala/hmda/api/http/InstitutionsHttpApi.scala

```diff
@@ -0,0 +1,198 @@
+package hmda.api.http
+
+import java.time.Instant
+
+import akka.Done
+import akka.actor.{ ActorRef, ActorSelection, ActorSystem }
+import akka.event.LoggingAdapter
+import akka.stream.ActorMaterializer
+import akka.pattern.ask
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.model.Multipart.BodyPart
+import akka.http.scaladsl.model._
+import akka.stream.scaladsl.{ Framing, Sink }
+import akka.util.{ ByteString, Timeout }
+import hmda.api.model._
+import hmda.persistence.{ FilingPersistence, HmdaFileUpload, SubmissionPersistence }
+import hmda.persistence.FilingPersistence.GetFilingByPeriod
+import hmda.persistence.HmdaFileUpload.{ AddLine, _ }
+import hmda.persistence.InstitutionPersistence.GetInstitutionById
+import hmda.persistence.SubmissionPersistence.{ CreateSubmission, GetLatestSubmission }
+import hmda.api.protocol.processing.{ FilingProtocol, InstitutionProtocol }
+import hmda.model.fi.{ Filing, Institution, Submission }
+import hmda.persistence.CommonMessages._
+import hmda.persistence.{ CommonMessages, FilingPersistence, SubmissionPersistence }
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Failure, Success }
+import spray.json._
+
+trait InstitutionsHttpApi extends InstitutionProtocol {
+
+  implicit val system: ActorSystem
+  implicit val materializer: ActorMaterializer
+  val log: LoggingAdapter
+
+  implicit val timeout: Timeout
+
+  val splitLines = Framing.delimiter(ByteString("\n"), 2048, allowTruncation = true)
+
+  val institutionsPath =
+    path("institutions") {
+      val institutionsActor = system.actorSelection("/user/institutions")
+      get {
+        val fInstitutions = (institutionsActor ? GetState).mapTo[Set[Institution]]
+        onComplete(fInstitutions) {
+          case Success(institutions) =>
+            complete(ToResponseMarshallable(Institutions(institutions)))
+          case Failure(error) =>
+            log.error(error.getLocalizedMessage)
+            complete(HttpResponse(StatusCodes.InternalServerError))
+        }
+      }
+    }
+
+  val institutionByIdPath =
+    path("institutions" / Segment) { fid =>
+      extractExecutionContext { executor =>
+        val institutionsActor = system.actorSelection("/user/institutions")
+        val filingsActor = system.actorOf(FilingPersistence.props(fid))
```

Comment:
As a reminder from when we talked yesterday, remember to shut down this actor after the call finishes.